proto/v1/versions/structure_versions.proto

```diff
@@ -0,0 +1,151 @@
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the version of the proto structures for explorations messages. This structure is never
+// expected to change.
+//
+// See the README for details on how these versions are meant to be used.
+message ExplorationProtoVersion {
+  option (structure_proto_version_type) = UNVERSIONED_STRUCTURE_PROTO;
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the version of the proto structures for question messages. This structure is never
+// expected to change.
+//
+// See the README for details on how these versions are meant to be used.
+message QuestionProtoVersion {
+  option (structure_proto_version_type) = UNVERSIONED_STRUCTURE_PROTO;
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the version of the proto structures for state messages.
+// This structure is never expected to change.
+//
+// Note that this version also implies compatibility among changes to interactions since
+// interactions are defined using strong types, and all of those proto messages are bound to this
+// version.
+//
+// See the README for details on how these versions are meant to be used.
+message StateProtoVersion {
+  option (structure_proto_version_type) = UNVERSIONED_STRUCTURE_PROTO;
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the proto version of multiple language-tied messages. Note that these are versioned
+// separately from the structures that reference them since multiple topic substructures reference
+// common messages (e.g. RecordedVoiceovers), so changes in the structures need to be properly
+// shared across all use cases without necessarily updating the individual proto versions for each
+// of those messages. Note also that this structrure is never expected to change.
+//
+// See also the README for details on how these versions are meant to be used.
+message LanguageProtosVersion {
+  option (structure_proto_version_type) = UNVERSIONED_STRUCTURE_PROTO;
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the version of the proto structures for image messages. This structure is never
+// expected to change. This also shares some characteristics with LanguageProtosVersion in that it's
+// used to track versioning for highly centralized & shared structures.
+//
+// See the README for details on how these versions are meant to be used.
+message ImageProtoVersion {
+  option (structure_proto_version_type) = UNVERSIONED_STRUCTURE_PROTO;
+
+  // The numeric version corresponding to the containing proto's structure.
+  uint32 version = 1;
+}
+
+// Represents the type of versioning corresponding to a structure proto message.
+enum StructureProtoVersionType {
```

Comment:
I think this could use more explanation -- what is a 'version type'? What is a "structure proto'?

Maybe better to talk about "versioning system"? Types already mean something else so "version type" is a bit confusing (basically it scans as "int" since versions are typically integers).